### ISSUE TITLE
Is it differentiable to render silhouette of mesh?

### ISSUE THREAD

**Issue by xhuan8** [facebookresearch/pytorch3d#730]
I try to use silhouette to supervise the training of network, but it does not converge,
is this process differentiable? 

`teapot_mesh = Meshes(verts=vertices, faces=faces)
                            cameras = PerspectiveCameras(device=device, 
                                    focal_length=((config.FOCAL_LENGTH, config.FOCAL_LENGTH),),  # (fx_screen, fy_screen)
                                    principal_point=((proxy_rep_input_wh/2, proxy_rep_input_wh/2),),  # (px_screen, py_screen)
                                    image_size=((proxy_rep_input_wh, proxy_rep_input_wh),),  # (imwidth, imheight)
                                    )
                            blend_params = BlendParams(sigma=1e-4, gamma=1e-4)
                            raster_settings = RasterizationSettings(
                                image_size=proxy_rep_input_wh, 
                                #blur_radius = 0.0,
                                blur_radius=np.log(1. / 1e-4 - 1.) * blend_params.sigma, 
                                faces_per_pixel=100, 
                            )
                            #print(np.log(1. / 1e-4 - 1.) * blend_params.sigma)
                            silhouette_renderer = MeshRenderer(
                                rasterizer=MeshRasterizer(
                                    cameras=cameras, 
                                    raster_settings=raster_settings
                                ),
                                shader=SoftSilhouetteShader(blend_params=blend_params)
                                )
        
                            sil_image = silhouette_renderer(meshes_world=teapot_mesh, T=translation, R=cam_R)
                            sil_image = sil_image[...,3]`

`silhouette_loss = torch.abs(sil_image  - label)`

**Comment by xhuan8** [comment]
seems there is something wrong with my data
